src/components/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState, useMemo } from 'react';
+
+import * as postService from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  children: React.ReactNode;
+};
+
+interface TodoContextType {
+  USER_ID: number,
+  todos: Todo[],
+  tempTodo: Todo | null,
+  filteredTodos: Todo[],
+  errorMessage: string,
+  isSelected: string,
+  isLoading: null | number,
+  addTodo: (newTodo: Todo) => void,
+  filterTodo: (filter: string) => void,
+  deleteTodo: (todoId: number) => void
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+  setIsSelected: React.Dispatch<React.SetStateAction<string>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<number | null>>
```

Comment:
pls rename this state because now it sounds like boolean value but in fact it is an id of loading todo